src/test/java/com/google/sps/SessionContextTest.java

```diff
@@ -35,24 +35,36 @@ public void getLoggedInUser_ReturnUser() {
         com.google.appengine.api.users.User googleUser = new com.google.appengine.api.users.User("email", "domain");
         when(userService.getCurrentUser()).thenReturn(googleUser);
         when(userRepository.getUser(googleUser)).thenReturn(testUser);
-
-        User resultUser = sessionContext.getLoggedInUser();
-
-        Assert.assertEquals(testUser, resultUser);
+        
+        String resultUserName = "";
+        try {
+            User resultUser = sessionContext.getLoggedInUser();
+            resultUserName = resultUser.getName();
+        } catch(Exception e) {
+            Assert.fail("Exception caught" + e);
+        }
+
+        Assert.assertEquals(testUser.getName(), resultUserName);
     }
 
+   @Test
+   public void getLoggedInUser_ReturnNull() {
+       User testUser = null;
+       com.google.appengine.api.users.User googleUser = null;
+       when(userService.getCurrentUser()).thenReturn(googleUser);
+       
+       String resultUserName = "";
+        try {
+            User resultUser = sessionContext.getLoggedInUser();
+            if(resultUser == null) {
+                Assert.assertEquals(testUser, resultUser);
+            }
+        } catch(Exception e) {
+            System.err.println("Exception caught " + e);
```

Comment:
Since you don't expect exceptions here, can you add an assertion failure here (L63) so the test fails if an exception is caught?

Reply:
Yea.

Reply:
Done

Reply:
Done